Tests/src/artofillusion/object/ObjectInfoTest.java

```diff
@@ -0,0 +1,852 @@
+    
+    ObjectInfo source = new ObjectInfo(sourceGeometry, sourceCoords, "Source");
+    source.setVisible(false);
+    source.setLocked(true);
+    source.setId(100);
+    ObjectInfo target = new ObjectInfo(new Sphere(1d,1d,1d), new CoordinateSystem(), "Target");
+    
+    target.copyInfo(source);
+    
+    assertEquals(100, target.getId());
+    assertEquals("Source", target.getName());
+    assertEquals(sourceGeometry, target.getObject());
+    assertEquals(sourceCoords, target.getCoords());
+    
+    assertTrue(target.isLocked());
+    assertFalse(target.isVisible());
+    
+  }
+  
+  /**
+   * Test objectInfo copy data from other objectInfo and points to same geometry
+   * Checks that source empty tracks overwrite existed one
+   */  
+  public void testCopyInfoWithEmptyTracksOverExisted()
+  {
+    ObjectInfo source = new ObjectInfo(new Cube(1d,1d,1d), new CoordinateSystem(), "Source");
+    
+    ObjectInfo target = new ObjectInfo(new Sphere(1d,1d,1d), new CoordinateSystem(), "Target");
+    target.addTrack(new PositionTrack(target), 0);
+    target.addTrack(new RotationTrack(target), 1);
+    
+    target.copyInfo(source);
+    assertNull(target.getTracks());
+    
+    
+  }
+  
+  /**
+   * Test objectInfo copy data from other objectInfo and points to same geometry
+   * Checks that source tracks overwrite existed one
+   */  
+  @Test  
+  public void testCopyInfoWithTracksOverExisted()
+  {
+    ObjectInfo source = new ObjectInfo(new Cube(1d,1d,1d), new CoordinateSystem(), "Source");
+    source.addTrack(new TextureTrackImpl((source)), 0);
+    
+    ObjectInfo target = new ObjectInfo(new Sphere(1d,1d,1d), new CoordinateSystem(), "Target");
+    target.addTrack(new PositionTrack(target), 0);
+    target.addTrack(new RotationTrack(target), 1);
+    
+    target.copyInfo(source);
+    assertNotNull(target.getTracks());
+    assertEquals(1, target.getTracks().length);
+    Track testT = target.getTracks()[0];
+    assertTrue(testT instanceof TextureTrack);
+    assertEquals(target, target.getTracks()[0].getParent());
+  }
+  
+  /**
+   * Test objectInfo copy data from other objectInfo and points to same geometry
+   * Checks that source distortion copied
+   */ 
+  @Test
+  public void testCopyInfoWithDistortionOverEmpty()
+  {
+    ObjectInfo source = new ObjectInfo(new Cube(1d,1d,1d), new CoordinateSystem(), "Source");
+    Distortion dist = new DistortionImpl();
+    source.setDistortion(dist);
+    
+    ObjectInfo target = new ObjectInfo(new Sphere(1d,1d,1d), new CoordinateSystem(), "Target");
+    target.copyInfo(source);
+    assertNotNull(target.getDistortion());
+    assertTrue(target.getDistortion() instanceof DistortionImpl);
+    
+  }
+  
+  /**
+   * Test objectInfo copy data from other objectInfo and points to same geometry
+   * Checks that empty distortion data overwrites existed one
+   */ 
+  @Test
+  public void testCopyInfoWithNullDistortionOverExisted()
```

Comment:
This test fails with the current codebase. There is an explicit check to make sure that the distortion is _not_ null before copying it over existing data. See:

https://github.com/ArtOfIllusion/ArtOfIllusion/blob/7623fc4a7d3d58434c90b294ad499479f210d55c/ArtOfIllusion/src/artofillusion/object/ObjectInfo.java#L136-L139

@peastman do you know what the reason is here for _not_ overwriting with a null?

Reply:
No reason I can think of.  I suspect I just forgot to handle the case where it's null.

Reply:
Makes sense to me.